dashboard/app/models/pd/workshop.rb

```diff
@@ -361,10 +363,33 @@ def self.send_reminder_to_close
     raise "Failed to send reminders: #{errors.join(', ')}" unless errors.empty?
   end
 
+  # Send follow up email to teachers that attended CSF Intro workshops which ended exactly X days ago
+  def self.send_follow_up_in_days(days)
+    # Collect errors, but do not stop batch. Rethrow all errors below.
+    errors = []
+
+    scheduled_end_in_days(-days).each do |workshop|
+      next unless workshop.course == COURSE_CSF && workshop.subject == SUBJECT_CSF_101
+      attedend_teachers = workshop.attending_teachers
+
+      workshop.enrollments.each do |enrollment|
+        next unless attedend_teachers.include?(enrollment.user)
+
+        email = Pd::WorkshopMailer.teacher_follow_up(enrollment)
+        email.deliver_now
+      rescue => e
+        errors << "teacher enrollment #{enrollment.id} - #{e.message}"
```

Comment:
Flattening these errors into one exception is good for the runtime behavior here, but may make it less obvious in our error tracking system when many mailers are failing.  One option is to use [`Honeybadger.notify`](https://docs.honeybadger.io/ruby/getting-started/reporting-errors.html) in the rescue block to surface these individual errors to the reporting system without raising.  I'd still raise, below, but I'm not sure you need to include all the error messages there.

Reply:
Good idea, I will add Honeybadger